### PR TITLE
Add option to set background in Unity8-dash

### PR DESCRIPTION
**Pull request by mariogrip**

### CODE REVIEW

File: qml/Dash/ScopesListCategoryItem.qml [hunk 1]
```diff
@@ -73,6 +73,7 @@ AbstractButton {
                 wrapMode: Text.Wrap
                 maximumLineCount: 1
                 verticalAlignment: Text.AlignHCenter
+		color: scopeStyle ? scopeStyle.foreground : theme.palette.normal.baseText
```
Comment (mardy) [inline qml/Dash/ScopesListCategoryItem.qml]: Don't use tabs :-)

File: qml/Dash/ScopesListCategoryItem.qml [hunk 2]
```diff
@@ -83,6 +84,7 @@ AbstractButton {
                 maximumLineCount: 1
                 verticalAlignment: Text.AlignHCenter
                 visible: text != ""
+		color: scopeStyle ? scopeStyle.foreground : theme.palette.normal.baseText
```
Comment (mardy) [inline qml/Dash/ScopesListCategoryItem.qml]: same here

File: qml/Dash/DashBackground.qml [hunk 3]
```diff
@@ -15,7 +15,24 @@
...
 
 import QtQuick 2.4
+import QtQuick.Window 2.2
+import AccountsService 0.1
+import ImageCache 0.1
+import GSettings 1.0
 
 Rectangle {
-    color: "white"
+	GSettings {
```
Comment (mardy) [inline qml/Dash/DashBackground.qml]: use 4 spaces, not tabs

File: qml/Dash/DashBackground.qml [hunk 4]
```diff
@@ -15,7 +15,24 @@
  */
 
 import QtQuick 2.4
+import QtQuick.Window 2.2
```
Comment (mardy) [inline qml/Dash/DashBackground.qml]: is this needed?

File: qml/Dash/DashBackground.qml [hunk 5]
```diff
@@ -15,7 +15,24 @@
...
+import AccountsService 0.1
+import ImageCache 0.1
```
Comment (mardy) [inline qml/Dash/DashBackground.qml]: and this?

File: plugins/Dash/ScopeStyle.qml [hunk 6]
```diff
@@ -26,17 +28,22 @@ import Ubuntu.Components 1.3
...
+    property var gsettings: GSettings {
+          schema.id: "com.ubuntu.touch.system-settings"
+      }
+
     /// Style object passed from the scope
     property var style: Object()
 
     /// Color used for text and symbolic icons
-    readonly property color foreground: "foreground-color" in style ? style["foreground-color"] : d.defaultDark
+    property color foreground: gsettings.dashBackground ? "white" : d.defaultDark
```
Comment (mardy) [inline plugins/Dash/ScopeStyle.qml]: why was "readonly" removed?
Then, with change, scopes are no longer able to chose their preferred background colour.
Maybe you wanted to write:
```
readonly property color foreground: gsettings.dashBackground ? "white" : ("foreground-color" in style ? style["foreground-color"] : d.defaultDark)
```
Though I wonder, why force white color? What if the background image is mostly white?

File: qml/Dash/ScopesListCategoryItem.qml (hunk 1 above)
Comment (mariogrip) [inline qml/Dash/ScopesListCategoryItem.qml]: Sorry, I'm used to atom "translating" tabs to spaces, will fix it.

File: plugins/Dash/ScopeStyle.qml (hunk 6 above)
Comment (mariogrip) [inline plugins/Dash/ScopeStyle.qml]: Removed readonly so colors can be changed on gsetting change, if not you have to restart unity8-dash to get the new style changes

**Pull request merged**